BackendCommands.php

```diff
@@ -134,43 +118,162 @@ public function updateCode()
+     *
+     * @hook post-command backend:config-export
+     *
+     * @param $result
+     * @param \Consolidation\AnnotatedCommand\CommandData $commandData
+     */
+    public function postConfigExportCommand($result, CommandData $commandData)
+    {
+        $exportedFiles = $this->getConfigFilesInDirectory($this->siteConfigSyncDirectory());
+        $overrideFiles = $this->getConfigFilesInDirectory($this->siteConfigOverrideDirectory());
+        $sharedFiles = $this->getConfigFilesInDirectory($this->configSharedDirectory());
+        $localFiles = $this->getConfigFilesInDirectory($this->siteConfigEnvironmentDirectory('local'));
+        $modifiedFiles = [];
+
+        foreach ($exportedFiles as $fileName => $fullPath) {
+            // First check, if the file should be put into the override directory.
+            // otherwise check, if the file should be put into the shared directory.
+            if (isset($overrideFiles[$fileName])) {
+                if (!$this->filesAreEqual($overrideFiles[$fileName], $fullPath)) {
+                    $this->filesystem->copy($fullPath, $overrideFiles[$fileName], true);
+                    $modifiedFiles[$fileName] = $fullPath;
+                }
+            } elseif (isset($sharedFiles[$fileName])) {
+                if (!$this->filesAreEqual($sharedFiles[$fileName], $fullPath)) {
+                    $this->filesystem->copy($fullPath, $sharedFiles[$fileName], true);
+                    $modifiedFiles[$fileName] = $fullPath;
+                }
+            }
+        }
 
-        if ($this->environment !== 'local') {
-            // Nimbus will overwrite local config files with the production values.
-            $this->process(['git', 'checkout', $this->siteConfigSyncDirectory().'/../local']);
+        // Give information to the user, when we modified a configuration that also exists in local config.
+        // TODO: revisit, when local config folder has been removed.
+        foreach ($modifiedFiles as $fileName => $fullPath) {
+            if (isset($localFiles[$fileName])) {
+                $this->io()->block('Configuration file "'.$fileName.'" was changed and exists in local config folder. Please check, if local config has to be manually modified.', 'INFO', 'fg=yellow');
+            }
         }
 
-        // Move config into shared and site specific folders.
-        // @todo Fix sync-config.sh expecting relative path.
-        $this->process(
-            ['scripts/sync-config.sh', $this->siteConfigSyncDirectory().'/../export'],
-            $this->projectDirectory()
-        );
+        // Remove files from override, that were not exported anymore.
+        foreach ($overrideFiles as $fileName => $fullPath) {
+            if (!isset($exportedFiles[$fileName])) {
+                $this->filesystem->remove($fullPath);
+            }
+        }
+
+        // Remove files from shared, that were not exported anymore.
+        foreach ($sharedFiles as $fileName => $fullPath) {
```

Comment:
When deleting from shared config, there should be warning, telling the user to check if configuration needs to be moved to other instance overridden directory.

Reply:
Good idea